identity/src/main/java/com/android/identity/util/ApplicationData.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.identity.util;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+/**
+ * An interface to an object storing application data in a key-value pair manner.
+ *
+ * <p>This interface exists to support applications which wish to store additional data it wants to
+ * associate with a credential/an object. For example, on a
+ * {@link com.android.identity.credential.Credential} object one could use this to store the
+ * document type (e.g. DocType for 18013-5 credentials), user-visible name, logos/background, state,
+ * and so on.
+ */
+public interface ApplicationData {
+
+    /**
+     * Sets application specific data.
+     *
+     * <p>Use {@link #getData(String)} to read the data back.
+     *
+     * @param key   the key for the data.
+     * @param value the value or {@code null} to remove.
+     * @return      the modified {@link ApplicationData}.
+     */
+    @NonNull ApplicationData setData(@NonNull String key, @Nullable byte[] value);
+
+    /**
+     * Sets application specific data as a string.
+     *
+     * <p>Like {@link #setData(String, byte[])} but encodes the given value
+     * using <a href="http://cbor.io/">CBOR</a> before storing it.
+     *
+     * @param key   the key for the data.
+     * @param value the value
+     * @return      the modified {@link ApplicationData}.
+     */
+    @NonNull ApplicationData setString(@NonNull String key, @NonNull String value);
+
+    /**
+     * Sets application specific data as a {@code long}.
+     *
+     * <p>Like {@link #setData(String, byte[])} but encodes the given value
+     * using <a href="http://cbor.io/">CBOR</a> before storing it.
+     *
+     * @param key   the key for the data.
+     * @param value the value.
+     * @return      the modified {@link ApplicationData}.
+     */
+    @NonNull ApplicationData setNumber(@NonNull String key, long value);
+
+    /**
+     * Sets application specific data as a {@link Timestamp}.
+     *
+     * <p>Like {@link #setData(String, byte[])} but encodes the given value
+     * using <a href="http://cbor.io/">CBOR</a> before storing it.
```

Comment:
Yup, can confirm that `SimpleApplicationData#setTimestamp` just stores the number of milliseconds since Epoch. And yeah, that makes sense so I'll remove the `Timestamp`-specific functions